### PR TITLE
Remove unused build pipelines. 

### PR DESCRIPTION
**Pull request by benjamin-brady**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
To remove old unused build pipelines as part of a security cleanup.

**Special notes for your reviewer**:
Removes the following pipelines:
- _UVM Build 1ES_ - Last run Dec 2023
- _[Test Single VHD] AKS Linux VHD Build - Msft Tenant_ - last run Feb 2022

See ADO Task:28144237 for more details.

### CODE REVIEW

**Review by djsly — APPROVED** [review]

File: .pipelines/.vsts-vhd-builder.yaml
Comment (cameronmeissner) [inline .pipelines/.vsts-vhd-builder.yaml:1]: this file is the source of our VHD build check-in pipeline used throughout this repo, and thus is still actively used

File: .pipelines/.vsts-kata-uvm-builder.yaml
Comment (cameronmeissner) [inline .pipelines/.vsts-kata-uvm-builder.yaml:1]: please confirm with Martin Heberling that this pipeline is no longer needed

**Review by cameronmeissner — CHANGES_REQUESTED** [review]
please see comments

**Comment by cameronmeissner** [comment]
confirmed with Anuj that the UVM build is safe to remove, though we'll need to keep the other to preserve the PR check-in. I think for that one you can just go ahead and delete the ADO pipeline itself

File: .pipelines/.vsts-vhd-builder.yaml
Comment (anujmaheshwari1) [reply .pipelines/.vsts-vhd-builder.yaml:1]: @benjamin-brady this file is the base template for any pipeline building VHDs. we just want to delete the pipeline itself, not this yaml, which is used by other pipelines still building VHDs.

File: .pipelines/.vsts-kata-uvm-builder.yaml
Comment (anujmaheshwari1) [reply .pipelines/.vsts-kata-uvm-builder.yaml:1]: please only delete the pipeline, and please confirm that the UVM pipeline Martin linked here https://teams.microsoft.com/l/message/19:5ff880aaae474fcba19cbd69b5e9a9ec@thread.skype/1718208183667?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=e121dbfd-0ec1-40ea-8af5-26075f6a731b&parentMessageId=1718162069240&teamName=Azure%20Container%20Compute&channelName=AKS%20Node%20SIG&createdTime=1718208183667 doesn't use the same yaml

Comment (benjamin-brady) [reply .pipelines/.vsts-kata-uvm-builder.yaml:1]: @anujmaheshwari1 The kata-build-uvm pipeline Martin linked is using a different yaml: `aks-rp  /.pipelines/agentbaker/kata-build-uvm.yaml`. He is okay with deleting the ADO and this `.vsts-kata-uvm-builder.yaml` file is just placeholder code that doesn't do anything anyway.

Is it safe to delete this`.vsts-kata-uvm-builder.yaml` then? If not, there will be no yaml files to delete and I'll abandon this PR.

**Comment by benjamin-brady** [comment]
I have removed changes to the VHD builder yaml and I believe the UVM build is safe to remove since it's just placeholder code and the ADO pipeline pointing to it hasn't been run in 6 months.

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**